test/manipulation.test.js

```diff
@@ -2079,6 +2079,40 @@ describe('manipulation', function() {
       }
     });
 
+    describe('custom value providers', function() {
+      var ModelWithCustomProvider, modelInstance;
+      before(modelWithValueProvider);
+
+      it('uses user-registered value providers', function() {
+        var RAND_REGEXP = /^[0-9a-f]+$/i;
+        modelInstance.rand.should.match(RAND_REGEXP);
+      });
+
+      it('passes the model instance into the value provider', function() {
+        modelInstance.slugId.should.eql('test-instance');
+      });
+
+      function modelWithValueProvider(cb) {
+        ModelWithCustomProvider = db.define('ModelWithCustomProvider', {
```

Comment:
I think the db.define call's name might be what's too long.
`db.define('ShortenMe', { ... })`